### ISSUE TITLE
Implement new subtle notification banner in KMW dashboard area

### ISSUE THREAD

**Issue by zutigrm** [google/site-kit-wp#9371]
## Feature Description

The `ACRSubtleNotification` component should be added to the new highest priority widget added to the `AREA_MAIN_DASHBOARD_KEY_METRICS_PRIMARY` widget area in #9345 . 

The `ACRSubtleNotifciation` component should be adapted for showing within `VIEW_CONTEXT_MAIN_DASHBOARD` context, to include different content variation, and as part of this issue, banner should be surfaced in the dashboard within key metrics widget area only for users who already setup KMW using tailored metrics, when initial ACR events are detected - and those events match to they goals. And CTA should offer them to automatically include new ACR widgets based on the detected events

Figma design can be found [here](https://www.figma.com/design/S2CNc1toIzC7UrGMgs79S6/Conversions?node-id=1965-15179&node-type=frame&t=Jv59rveRss6eCqpE-0)

Refer to the [Subtle banner notification for detected events](https://docs.google.com/document/d/1JvA3LK5SNfGH2o4OVnwjDdsAFuQMA9ZJj0T09RTqoy8/edit?pli=1&resourcekey=0-e90bcFGzAI6NhNu_D5i6hQ#heading=h.4vc3htg11p6z) section of the design doc

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* New banner component `ACRDashboardSubtleNotification` is created 
* New widget component `ACRNotificationCTAWidget` is included in the `keyMetricsEventDetectionCalloutNotification` widget added in [9345](https://github.com/google/site-kit-wp/issues/9345) replacing the `null` component
* Said component should surface in the widget area only if:
  * Initial conversion events are detected
  * And key metrics were setup using tailored metrics (answering the user questions)
  * And detected events match with their saved goals (there are metric tiles that are valuable to include, otherwise we will not show the banner)
* Said component should match the [Figma design](https://www.figma.com/design/S2CNc1toIzC7UrGMgs79S6/Conversions?node-id=1965-15179&node-type=frame&t=Jv59rveRss6eCqpE-0)
* Clicking the `Maybe later` CTA should dismiss the banner, and show the admin menu tooltip as per [design](https://www.figma.com/design/S2CNc1toIzC7UrGMgs79S6/Conversions?node-id=1773-22571&node-type=frame&t=CWAdwrNYTvlczYwL-0)
* Clicking the `Add metrics` CTA should automatically include additional ACR metrics based on the existing answer/preference
* Check the [Subtle banner notification for detected events](https://docs.google.com/document/d/1JvA3LK5SNfGH2o4OVnwjDdsAFuQMA9ZJj0T09RTqoy8/edit?pli=1&resourcekey=0-e90bcFGzAI6NhNu_D5i6hQ#heading=h.4vc3htg11p6z) section of the design doc
* Banner should be displayed only when `conversionReporting` feature flag is enabled

## Implementation Brief

* [ ] Update `assets/js/googlesitekit/datastore/user/constants.js`
  * Create a new constant that will map the event names with their widget slugs, example: `const CONVERSION_EVENTS_WIDGETS = [{'purchase': [KM_ANALYTICS_TOP_CITIES_DRIVING_PURCHASES, KM_ANALYTICS_TOP_TRAFFIC_SOURCE_DRIVING_PURCHASES]}, {...}]`
* [x] Add `assets/js/components/KeyMetrics/ACRDashboardSubtleNotification.js` component
  * You can re-use the classes from `ACRSubtleNotification` to keep the design/layout 
  * `Maybe later` CTA callback should trigger `dismissNewConversionReportingEvents` action from `MODULES_ANALYTICS_4` datastore, and open `adminMenuTooltip()` and match the [design](https://www.figma.com/design/S2CNc1toIzC7UrGMgs79S6/Conversions?node-id=1773-22571&node-type=frame&t=CWAdwrNYTvlczYwL-0) . You can see an example for this in `assets/js/components/KeyMetrics/KeyMetricsSetupCTAWidget.js`, and part of the logic can be implemented in the `ACRNotificationCTAWidget` component 
  * `Add metrics` CTA callback should update the `includeConversionTailoredMetrics` KM setting to `true` (and trigger the save so it is updated in the DB)
* [x] Add `assets/js/components/KeyMetrics/ACRNotificationCTAWidget.js` component
  *  It should accept `Widget` and `WidgetNull` props
  * You can see an example in `assets/js/components/KeyMetrics/KeyMetricsSetupCTAWidget.js`
  * It should render `ACRDashboardSubtleNotification` component if following conditions are met (otherwise it should render `WidgetNull`):
    * Banner should show if `isUserInputCompleted` selector from `CORE_USER` store is true
    * And `hasNewConversionReportingEvents` selector is not falsy - even it is included in `isActive` check above to return early if this condition is not met, we still need it here, as this will be expanded to include lost events detection in other issue, so `isActive` check will still render the widget if either new or lost events are detected at that point
      * Pull the initially detected events using `getDetectedEvents` setting from Analtyics module and compare if `detectedEvents` holds the same value as `hasNewConversionReportingEvents`. They will match only when initial set of events is detected, and verify that `includeConversionTailoredMetrics` analytics setting is not true - so if user already used CTA to include new events, and at some point lost all events and now they have been detected again, we want to skip showing this banner variation
   * And to verify that new detected events are matching the selected purpose, use `getAnswerBasedMetrics` which was updated in [9438](https://github.com/google/site-kit-wp/issues/9438) to accept `purpose` param. Retrieve purpose answer from `getUserInputSettings` selector and extract `purpose` answer https://github.com/google/site-kit-wp/blob/5acfa2a08cadff1f0cf5153e186331d29817c293/assets/js/components/KeyMetrics/ConfirmSitePurposeChangeModal.js#L59. This will return the list of KMW slugs that are included for that purpose, including ACR widget that would be added. Pull the list from `KM_ANALYTICS_TOP_CITIES_DRIVING_PURCHASES` and retrieve the widget slugs belonging to the detected events (from `getDetectedEvents`). If widgets from one of the events are present in purpose related widgets, then this check should be considered as `true` for showing the banner along the 2 previous checks. 
   * Following the above conditions, include the content variation matching this banner in [Figma](https://www.figma.com/design/S2CNc1toIzC7UrGMgs79S6/Conversions?node-id=1965-15179&node-type=frame&t=Jv59rveRss6eCqpE-0), by passing it as props to the `ACRDashboardSubtleNotification` component
* [x] In `assets/js/googlesitekit/widgets/register-defaults.js`
  * Include `ACRNotificationCTAWidget` component to the `keyMetricsEventDetectionCalloutNotification` widget
  * Leverage `isActive` prop to prevent it from rendering early if basic conditions are not met:
    * Return early if `hasNewConversionReportingEvents` selector from `MODULES_ANALYTICS_4` does not have any data 

### Test Coverage

* Add basic test coverage for `ACRNotificationCTAWidget` and stories for `ACRDashboardSubtleNotification`

## QA Brief
Note: Tester plugin has not been updated yet, but contact @zutigrm for "under the counter" early version that can be installed manually for purpose of testing this issue

* Setup Site Kit with Analytics Module
* Callout will surface or in following scenario:
  * When user input is completed without events being detected (use `Publish blog` as site purpose answer)
  * New event is detected for the first time
    * But the detected event, should be matching the metrics that will be included in the current site purpose, in this case for publishing blog it would be `contact`, `generate_lead` or `submit_lead_form` events. Any other event, should not surface the callout
* In other scenarios banner should not surface, including when KMW is setup manually, or when events for the site purpose are already detected during user input completion. In that case all ACR KMW will be included, and banner should not surface
* Some potential scenarios for testing:
* **Scenario 1**
  * Setup user input using `Publish blog` site purpose answer, without events being detected.
  * Force detection of the different event, for example `add_to_cart`, banner should not surface 
* **Scenario 2**
  *  Setup user input same way as in scenario 1, only force event detection of `contact` event
  * Upon completing the questionnaire, all ACR should be included and banner should not surface
* **Scenario 3**
  * Setup user input using `sell products` answer - as this one is specific comparing to all other
  * No event should be detected when doing setup
  * Force detection of `add_to_cart` or `purchase` events, but not both of them together
  * Banner should appear. Click `Add metrics` CTA, it should include only add to cart or purchase related metrics, depending which event you included, both not both
  * Then check with both events detected - it should include all ACR metrics for both events
* **Scenario 4**
  * Setup user input using `Monetize content (with ads or affiliate links)` site purpose
  * This site purpose does not have any ACR metric associated with it, so any event forced as detected should not surface the banner
* **Scenario 5**
  * Setup user input using `Sell products` site purpose, for example, with no events detected initially
  * Switch to manual selection - by replacing existing metric(s), orr adding one or more, and saving  it though the selection panel
  * Force detection of `add_to_cart` event which would trigger the banner, but since we switched to manual selection - banner should not surface
* In all scenarios - `Add metrics` should include ACR metrics, and banner should disappear
* Maybe later CTA will dismiss the banner 

## Changelog entry

* Implement new subtle notification banner in KMW dashboard area.

**Comment by 10upsimon** [comment]
@zutigrm AC ✅ Moving to IB

**Comment by eugene-manuilov** [comment]
@zutigrm, I think we need to create a completely separate component instead of reusing `ACRSubtleNotification` because there is nothing that we can reuse here but its design. Let's create a new component and use class names from that component.

**Comment by zutigrm** [comment]
Thanks @eugene-manuilov , I though initially to avoid duplicating it, but it will reduce on the code complexity if we include another component. IB updated

**Comment by eugene-manuilov** [comment]
Thanks, @zutigrm. A few more comments from me.
 
> * Include `ACRDashboardSubtleNotification` component to the `keyMetricsEventDetectionCalloutNotification` widget

Instead of using the notification component directly, we need to create a new widget component that will render the notification based on defined conditions. The notification component itself should only take care of the design and callback functions for its buttons.

**Comment by tofumatt** [comment]
I don't see anything in this issue that is interacting with the new Banner Notifications datastore, which is the unified way for the plugin to manage notifications going forward.

I would've expected this to use https://github.com/google/site-kit-wp/blob/f61b74fbb94c09bd334d508efb97ebf7d4c9cd1f/assets/js/googlesitekit/notifications/datastore/notifications.js#L62 like any other notification 🤔

The point of that epic was to create all notifications in the same way so we could keep track of them and render them appropriately (eg. no duplicates in the same region, etc.)

I think this issue (and #9345) need to be re-thought. It seems like we want to add a new notification area to https://github.com/google/site-kit-wp/blob/f61b74fbb94c09bd334d508efb97ebf7d4c9cd1f/assets/js/googlesitekit/notifications/datastore/constants.js#L26-L30, since this component looks like and acts like a notification, right?

cc @jimmymadon who worked on the refactoring of the notifications—am I off-base here and this shouldn't be a new notification area? 😅

**Comment by tofumatt** [comment]
I spoke to @jimmymadon about this on Slack and he mentioned that this was discussed in the Banner Notifications Refactoring Design Doc and it was decided not to register a whole section just for this one component/section.

I think it's worth reconsidering just to stay consistent, but that can be a later discussion. Apologies for the issue hijacking, ignore my previous comment for now 😄

**Comment by eugene-manuilov** [comment]
Thanks, @zutigrm. IB ✔

**Comment by benbowler** [comment]
Adding #9379 as a blocker because this ticket requires the `hasNewConversionReportingEvents` selector.

**Comment by benbowler** [comment]
Handing over and un-assigning for vacation, [all work so far is in this PR with details in the PR body](https://github.com/google/site-kit-wp/pull/9612), this is still blocked so it's worth waiting for the blocker to be merged.

**Comment by zutigrm** [comment]
I will continue working on it, as soon as blocker is merged

**Comment by zutigrm** [comment]
@adamdunnage There is a learn more [link](https://www.figma.com/design/S2CNc1toIzC7UrGMgs79S6/Conversions?node-id=1965-15179&node-type=frame&t=mLwWrFc0left2H2F-0) in the callout, that needs documentation url

**Comment by jimmymadon** [comment]
@adamdunnage There is another learn more link on #9335 too! Perhaps we should standardise these? c.c. @zutigrm

**Comment by kelvinballoo** [comment]
### QA Update ⚠ 

Tested this and I have a couple of observations:

**ITEM 1:**
There is no tooltip that appears after I dismissed the banner.

<Details>

https://github.com/user-attachments/assets/c1f9f5b2-fbbd-4d3a-8b73-39b72b581c65

</Details>
_____________________________________________________________________________________

**ITEM 2:**
On mobile implementation:
- the margins are currently 16 on the side and top. 
- Also, the buttons are centrally aligned.

On figma:
- it's 24px on the sides and 14px at the top. 
- The buttons are not centred. They are right-sided.

Should we align to Figma?

<Details>
Figma:

![Image](https://github.com/user-attachments/assets/9467d11b-b545-4c50-b9ee-fe314405ab71)

Current implementation:

![Image](https://github.com/user-attachments/assets/42bd8976-721e-44c4-86ea-12681ecb6c46)


</Details>

_____________________________________________________________________________________

**ITEM 3:**
For scenario 2, when the QAB says " all ACR should be included and banner should not surface".
Does that mean the ACR should only appear in the selection panel but it doesn't need to be selected?
Or should it be selected?

Currently, they are only showing up but not selected.
<Details>

https://github.com/user-attachments/assets/90669d63-a707-4ef7-848b-48f88e5d211c

</Details>

_____________________________________________________________________________________

**ITEM 4:**
For scenario 3, when I tested 'add to cart', the ACRs will appear in the selection panel:
- Top traffic source driving add to cart
- Top cities driving add to cart

However, only `Top traffic source driving add to cart` would be selected.
Should `Top cities driving add to cart` be selected as well?

<Details>
![Image](https://github.com/user-attachments/assets/c80d4b1c-2b8a-47db-bf95-7418402c9700)

</Details>

_____________________________________________________________________________________

**ITEM 5:**
Consider the following scenario:
- I do the tailored metrics 
- I force event 'add to cart'
- Banner appears and I add the metrics
- I force event 'purchase'
Upon refreshing the dashboard, should the banner appear again?
Also, I noticed that whenever we force an event in the tester plugin, it might take a bit of time before the banner appears. Is that expected? 

_____________________________________________________________________________________

**ITEM 6:**
Are we expected to test all the different site purpose for this issue per the attached screenshot? Or that will be under another ticket?
<Details>

![Image](https://github.com/user-attachments/assets/075dcc61-65e2-4eae-aae4-b43a20717909)

</Details>

Also, do we have a mapping as to which tester plugin event (e.g. contact) would make which specific ACR appear? I don't think I found this from the design doc.

**Comment by zutigrm** [comment]
@kelvinballoo Thanks,

> ITEM 1:
There is no tooltip that appears after I dismissed the banner.

That's correct behaviour, tooltip will not be included in this phase

> ITEM 2:
On mobile implementation:

> the margins are currently 16 on the side and top.
Also, the buttons are centrally aligned.
On figma:

> it's 24px on the sides and 14px at the top.
The buttons are not centred. They are right-sided.
Should we align to Figma?

Good spot, thanks. I will update this

> ITEM 3:
For scenario 2, when the QAB says " all ACR should be included and banner should not surface".
Does that mean the ACR should only appear in the selection panel but it doesn't need to be selected?
Or should it be selected?

> Currently, they are only showing up but not selected.

Let's sync on this, to verify event was loaded during setup, all metrics related to the detected event should be showing in both dashboard and selection panel as selected

> ITEM 4:
For scenario 3, when I tested 'add to cart', the ACRs will appear in the selection panel:

> Top traffic source driving add to cart
Top cities driving add to cart
However, only Top traffic source driving add to cart would be selected.
Should Top cities driving add to cart be selected as well?

This is correct, based on the tailored metrics table in design doc, only `Top traffic source driving add to cart` is included for Sell products purspose

> TEM 5:
Consider the following scenario:

> I do the tailored metrics
I force event 'add to cart'
Banner appears and I add the metrics
I force event 'purchase'
> Upon refreshing the dashboard, should the banner appear again?

Not in this issue, but due to some logic change it will appear after #9578 is merged

> Also, I noticed that whenever we force an event in the tester plugin, it might take a bit of time before the banner appears. Is that expected?

Yeah I had implemented some delay for testing purpose, but earlier today removed that, will send you updated plugin

> Are we expected to test all the different site purpose for this issue per the attached screenshot? Or that will be under another ticket?

Good question, basically, covering `publish blog` and `sell products` and the scenarios in QAB will suffice, since most of the purpose include only leads related metrics, expect for `sell products` and the monetisation which have different metrics for 2 events, and none of the ACR metrics respectively 

> Also, do we have a mapping as to which tester plugin event (e.g. contact) would make which specific ACR appear? I don't think I found this from the design doc.

Not at the moment, I will update design doc for this

**Comment by kelvinballoo** [comment]
Hi @zutigrm ,

**For ITEM 3,** 
I have a video showcasing the scenario with the ACRs not being selected.
<Details>

https://github.com/user-attachments/assets/a7ea1975-693c-4bed-8eb7-3b930a9f448f

</Details>

Assigning ticket to you to review ITEM 3 and fix ITEM 2.

**Comment by kelvinballoo** [comment]
### QA Update ✅ 

Thanks @zutigrm 

Noted on ITEM 1, 4 and 5.

ITEM 2 is verified good. ✅ 
On mobile, the sides are now 24px and 14px at the top.
The buttons are also right sided.

<Details>

![Image](https://github.com/user-attachments/assets/ae99f242-08bb-40da-a32a-2c9f2f10783c)

</Details>
_______________________________________________________________

ITEM 3 : I retested with the latest version of the tester plugin and it's working as expected. ✅ 

<Details>
https://github.com/user-attachments/assets/b0e3c019-2085-4f4b-a797-dc2fa05dcf80

</Details>

_______________________________________________________________

Other than that, everything was working as expected: ✅ 

- Scenarios 1-5 are working as defined in the QAB.
   - Said component matches figma and surfaces in the widget area only if:
      - Initial conversion events are detected
      - And key metrics were setup using tailored metrics
      - And detected events match with their saved goals
   - Clicking the Add metrics CTA automatically includes additional ACR metrics based on the existing answer/preference and banner disappears
   <Details>

   https://github.com/user-attachments/assets/78ef4196-b622-46e9-8d18-196116654615

   https://github.com/user-attachments/assets/d1dded6f-5b24-4a32-8964-a542520ac46e

   https://github.com/user-attachments/assets/fa7be465-14b5-4899-ac88-bceb99855531

   https://github.com/user-attachments/assets/92033245-e1c3-48b8-aac8-a2a638fa611b

   https://github.com/user-attachments/assets/9cbc9c46-81ca-45c9-8de3-1b39b6d1cc1c   
   
   https://github.com/user-attachments/assets/9dba83d2-e140-4d9c-baa4-8b9b8c18a2bf

   https://github.com/user-attachments/assets/2e1d8599-fb87-470d-a4af-78baa366fe95

   </Details>
 
- Maybe later CTA will dismiss the banner. Tooltip isn't appearing as it's not part of this issue.

   <Details>

   https://github.com/user-attachments/assets/187fadb1-ca18-414a-9e6e-8a36b1cb53d9

   </Details>
 
- Banner should be displayed only when conversionReporting feature flag is enabled
   <Details>

   https://github.com/user-attachments/assets/b952b14f-45aa-4fc5-a329-4b2c0e20cc9b

   </Details>
